cpp/tests/io/text/data_chunk_source_test.cpp

```diff
@@ -342,4 +342,33 @@ TEST_F(DataChunkSourceTest, BgzipCompressedSourceVirtualOffsets)
   test_source(input, *source);
 }
 
+TEST_F(DataChunkSourceTest, BgzipSourceVirtualOffsetsSingleCompressedGZipBlock)
+{
+  auto const filename = temp_env->get_temp_filepath("bgzip_source_offsets_single_compressed_block");
+  std::string const input{"collection unit brings"};
+  std::string head_garbage{"garbage"};
+  // make the head_garbage large enough so the local offset will overrun the compressed block size
+  for (int i = 0; i < 10; i++) {
+    head_garbage += head_garbage;
+  }
```

Comment:
```suggestion
  std::string head_garbage{"garbage"};
  head_garbage.reserve(7* (1<<10));
  // make the head_garbage large enough so the local offset will overrun the compressed block size
  for (int i = 0; i < 10; i++) {
    head_garbage += head_garbage;
  }
```
7 KB is not much, but if the string should be biggger, this might be efficient.

```
Benchmark                    Time             CPU   Iterations
--------------------------------------------------------------
BM_StringRepeat/10         393 ns          393 ns      1764275
BM_StringRepeat/20     5330959 ns      5330430 ns          131
BM_StringReserve/10        152 ns          151 ns      4567372
BM_StringReserve/20     572996 ns       572940 ns         1218
```

Reply:
feel free to ignore this comment.
address `begin_compressed_offset` comment.
Rest of the code looks good.

Reply:
good point, I preallocated the larger outputs